app/src/main/java/com/udacity/nanodegree/blooddonation/util/PermissionUtils.java

```diff
@@ -0,0 +1,43 @@
+package com.udacity.nanodegree.blooddonation.util;
+
+import android.Manifest;
+import android.app.Activity;
+import android.content.pm.PackageManager;
+import android.support.v4.app.ActivityCompat;
+import android.support.v4.content.ContextCompat;
+
+/**
+ * Created by tavishjain on 20-04-2018.
+ */
+
+public class PermissionUtils{
+
+    static Activity mActivity;
+
+    public PermissionUtils(Activity activity){
```

Comment:
One should never keep a reference of activity/context, as it dies pretty soon and leads to the crash problem.
Pass Context/Activity with the static function

Reply:
This code will lead to memory leak.
Remove static activity reference. Take it as an input Param.

Reply:
But @nipunjindal for the permissions, we require an instance of the activity. 

@ritesh-singh  could you please explain

Reply:
static life cycle - static variable life cycle is till the application lifecycle i.e variable exists until the application exits in memory.
mActivity will hold a reference to the activity which you are passing as a parameter and it will hold the activity reference till the application is in memory.  Even if the activity dies, mActivity will hold the activity reference & hence it will create a memory leak.

Reply:
For more info

Reply:
@tavishjain : pass it with requestLocationAndSms(Activity activity)
Don't hold any such references